samples/augment-reality-to-show-hidden-infrastructure/src/main/java/com/esri/arcgismaps/sample/augmentrealitytoshowhiddeninfrastructure/screens/MapScreen.kt

```diff
@@ -0,0 +1,163 @@
+ * Main screen layout for the sample app
+ */
+@Composable
+fun MapScreen(sampleName: String, locationPermissionGranted: Boolean, onNavigateToARScreen: () -> Unit) {
+
+    val mapViewModel: MapViewModel = viewModel()
+
+    // Initialize the location display with auto pan mode set to recenter
+    val locationDisplay = rememberLocationDisplay().apply {
+        setAutoPanMode(LocationDisplayAutoPanMode.Recenter)
+    }
+    var isViewmodelInitialized by remember { mutableStateOf(false) }
+    LaunchedEffect(isViewmodelInitialized) {
+        if (!isViewmodelInitialized && locationPermissionGranted) {
+            mapViewModel.initialize(locationDisplay)
+            isViewmodelInitialized = true
+        }
+    }
+
+    val isGeometryBeingEdited by remember { mutableStateOf(mapViewModel.isGeometryBeingEdited) }
+    val showElevationDialog by mapViewModel.showElevationDialog
+    val pipeInfoList = SharedRepository.pipeInfoList
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) }, content = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+            ) {
+                MapView(
+                    modifier = Modifier.fillMaxSize(),
+                    arcGISMap = mapViewModel.arcGISMap,
+                    locationDisplay = locationDisplay,
+                    geometryEditor = mapViewModel.geometryEditor,
+                    graphicsOverlays = listOf(mapViewModel.graphicsOverlay),
+                    onSingleTapConfirmed = {
+                        if (!mapViewModel.geometryEditor.isStarted.value) {
+                            mapViewModel.startPolylineEditing()
+                        }
+                    }
+                )
+                if (mapViewModel.statusText.value != "") {
+                    Box(
+                        modifier = Modifier
+                            .align(Alignment.TopCenter)
+                            .padding(8.dp)
+                            .clip(RoundedCornerShape(10.dp))
+                            .background(Color.Black.copy(alpha = 0.8f))
+                            .padding(16.dp)
+                    ) {
+                        Text(
+                            text = mapViewModel.statusText.value, color = Color.White
+                        )
+                    }
+                }
+                Box(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .padding(bottom = 24.dp),
+                    contentAlignment = Alignment.BottomCenter
+                ) {
+                    Column(
+                        horizontalAlignment = Alignment.CenterHorizontally
+                    ) {
+                        if (isGeometryBeingEdited.value) {
+                            Button(onClick = { mapViewModel.completePolyline() }) {
+                                Text("Complete polyline")
+                            }
```

Comment:
Should this sample also provide undoPipe / clearAll buttons to help with the drawing of these polylines? Swift enables buttons as vertices are added to the polyline.

Reply:
Users can navigate back out of the AR screen back to the Map screen, my perspective is why allow navigation back if the plotted geometry cannot be modified? The sample is to showcase adding pipes as hidden infrastructure, to me at the very least clearing the geometry to start over is relevant to the sample with the current sample navigation workflow.